src/objects/game-over.ts

```diff
@@ -0,0 +1,58 @@
+import { SnakeScene } from '../scenes/snake-scene';
+
+export class GameOver extends Phaser.GameObjects.Container {
+    snakeScene: SnakeScene;
+    SpaceKey: Phaser.Input.Keyboard.Key;
+    restartText: Phaser.GameObjects.Text;
+    DeadSnake;
+    visible = false;
+
+    constructor(snakeScene: SnakeScene) {
+        super(snakeScene);
+        this.snakeScene = snakeScene;
+    }
+
+    public preload() {
+        this.snakeScene.load.image('DeadSnake', "./assets/images/DeadSnake.png");
+        this.SpaceKey = this.snakeScene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.SPACE);
+    }
+
+    public isShown(): boolean {
```

Comment:
Tu peux remplacer toute cette méthode par:
`public isShown(): boolean {
 return this.visible;
}`